gax-java/gax/src/main/java/com/google/api/gax/tracing/MetricsTracerFactory.java

```diff
@@ -45,13 +47,31 @@
 public class MetricsTracerFactory implements ApiTracerFactory {
   protected MetricsRecorder metricsRecorder;
 
+  // These are client attributes and pertain every single trace
+  private final Map<String, String> attributes;
+
+  /** Creates a MetricsTracerFactory with no additional client level attributes. */
   public MetricsTracerFactory(MetricsRecorder metricsRecorder) {
+    this(metricsRecorder, ImmutableMap.of());
+  }
+
+  /**
+   * Pass in a Map of client level attributes which will be added to every single MetricsTracer
+   * created from the ApiTracerFactory.
+   */
+  public MetricsTracerFactory(MetricsRecorder metricsRecorder, Map<String, String> attributes) {
     this.metricsRecorder = metricsRecorder;
+    this.attributes = attributes;
   }
 
   @Override
   public ApiTracer newTracer(ApiTracer parent, SpanName spanName, OperationType operationType) {
-    return new MetricsTracer(
-        MethodName.of(spanName.getClientName(), spanName.getMethodName()), metricsRecorder);
+    MetricsTracer metricsTracer =
+        new MetricsTracer(
+            MethodName.of(spanName.getClientName(), spanName.getMethodName()), metricsRecorder);
+    for (Map.Entry<String, String> attributeEntrySet : attributes.entrySet()) {
+      metricsTracer.addAttributes(attributeEntrySet.getKey(), attributeEntrySet.getValue());
+    }
+    return metricsTracer;
```

Comment:
Possible options:
1. Use MetricTracer's `addAttributes()` to add *only* add client level attributes. The clientName and methodName attributes will continue to be passed via MethodName via the constructor.
2. Add MethodName attribute with `addAttributes()` and refactor `MetricsTracer` to only take in metricsRecorder

Probably going to keep it this way (option 1).